sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala

```diff
@@ -142,8 +143,13 @@ trait StreamingQuery {
   def processAllAvailable(): Unit
 
   /**
-   * Stops the execution of this query if it is running. This method blocks until the threads
-   * performing execution has stopped.
+   * Stops the execution of this query if it is running. This awaits the termination of the query
+   * execution threads or until the `spark.sql.streaming.stopTimeout` is hit. A timeout of 0
+   * milliseconds will block indefinitely.
```

Comment:
nit: explicitly say: By default stop will block indefinitely. You can make it timeout (ms) by setting the configuration...... (0 = block indefinitely).